bytecomp/bytelink.ml

```diff
@@ -603,7 +604,12 @@ let fix_exec_name name =
 let link objfiles output_name =
   let objfiles =
     if !Clflags.nopervasives then objfiles
-    else if !Clflags.output_c_object then "stdlib.cma" :: objfiles
+    else if !Clflags.output_c_object then
+      let objfiles =
+        if !Clflags.output_complete_executable then objfiles @ ["std_exit.cmo"]
+        else objfiles
+      in
+      "stdlib.cma" :: objfiles
     else "stdlib.cma" :: (objfiles @ ["std_exit.cmo"]) in
```

Comment:
This is a "take or leave" comment, but this is turning into a bit of an `if/let`-soup. Might it be better to calculate whether `std_exit.cmo` should be suffixed, calculate whether `stdlib.cma` should be prefixed and then compute `objfiles`?

Reply:
Yes, makes sense. Will amend.